process/block/sovereign/outgoingOperations.go

```diff
@@ -191,29 +195,30 @@ func (op *outgoingOperations) isSubscribed(event data.EventHandler, txHash strin
 	return false
 }
 
-func (op *outgoingOperations) getOperationData(event data.EventHandler) ([]byte, error) {
+func (op *outgoingOperations) getOperationData(event data.EventHandler) (dto.ChainID, []byte, error) {
 	evData, err := op.dataCodec.DeserializeEventData(event.GetData())
 	if err != nil {
-		return nil, err
+		return 0, nil, err
 	}
 
 	topics := event.GetTopics()
 	err = op.topicsChecker.CheckValidity(topics, evData.TransferData)
 	if err != nil {
-		return nil, err
+		return 0, nil, err
 	}
 
 	operation, err := op.createOperationData(topics, evData)
 	if err != nil {
-		return nil, err
+		return 0, nil, err
 	}
 
 	operationBytes, err := op.dataCodec.SerializeOperation(*operation)
 	if err != nil {
-		return nil, err
+		return 0, nil, err
 	}
 
-	return operationBytes, nil
+	// TODO: MX-16831 Here, we should have contracts emitting chain id events
```

Comment:
I don't think we need a different event for chain id.
We can maybe have `ChainID` in `evData`.